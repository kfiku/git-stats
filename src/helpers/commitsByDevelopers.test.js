// @ts-check

const commitsByDevelopers = require('./commitsByDevelopers')
const { commitsObjects, commitsObjectsMultiple } = require('./fixtures/git')

/* eslint-disable fp/no-unused-expression */
/* eslint-disable fp/no-nil */

describe('commitsByDevelopers', () => {
  it('single git log result', async () => {
    const result = commitsByDevelopers(commitsObjects)
    expect(result).toMatchSnapshot()
  })

  it('multiple git log result', async () => {
    const result = commitsByDevelopers(commitsObjectsMultiple)
    expect(result['kfiku.com@gmail.com'].commits.length).toBe(3)
    expect(result['john@localhost'].commits.length).toBe(1)
    expect(result['jan@localhost'].commits.length).toBe(2)

    expect(result['kfiku.com@gmail.com'].stats.filesChanged).toBe(3)
    expect(result['kfiku.com@gmail.com'].stats.insertions).toBe(9)
    expect(result['kfiku.com@gmail.com'].stats.deletions).toBe(4)

    expect(result['john@localhost'].stats.filesChanged).toBe(1)
    expect(result['john@localhost'].stats.insertions).toBe(0)
    expect(result['john@localhost'].stats.deletions).toBe(1)

    expect(result['jan@localhost'].stats.filesChanged).toBe(6)
    expect(result['jan@localhost'].stats.insertions).toBe(5)
    expect(result['jan@localhost'].stats.deletions).toBe(18)

    expect(result).toMatchSnapshot()
  })
})
