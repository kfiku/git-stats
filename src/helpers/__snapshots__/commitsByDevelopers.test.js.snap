// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commitsByDevelopers multiple git log result 1`] = `
Object {
  "jan@localhost": Object {
    "author": Object {
      "email": "jan@localhost",
      "name": "Jan Kowalski",
    },
    "commits": Array [
      Object {
        "authorEmail": "jan@localhost",
        "authorName": "Jan Kowalski",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "jk1",
        "stats": Object {
          "deletions": 3,
          "filesChanged": 4,
          "insertions": 1,
        },
      },
      Object {
        "authorEmail": "jan@localhost",
        "authorName": "Jan Kowalski",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "jk2",
        "stats": Object {
          "deletions": 15,
          "filesChanged": 2,
          "insertions": 4,
        },
      },
    ],
    "stats": Object {
      "deletions": 18,
      "filesChanged": 6,
      "insertions": 5,
    },
  },
  "john@localhost": Object {
    "author": Object {
      "email": "john@localhost",
      "name": "John Doe",
    },
    "commits": Array [
      Object {
        "authorEmail": "john@localhost",
        "authorName": "John Doe",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "jd1",
        "stats": Object {
          "deletions": 1,
          "filesChanged": 1,
          "insertions": 0,
        },
      },
    ],
    "stats": Object {
      "deletions": 1,
      "filesChanged": 1,
      "insertions": 0,
    },
  },
  "kfiku.com@gmail.com": Object {
    "author": Object {
      "email": "kfiku.com@gmail.com",
      "name": "Grzegorz Klimek",
    },
    "commits": Array [
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "1",
        "stats": Object {
          "deletions": 1,
          "filesChanged": 1,
          "insertions": 0,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "2",
        "stats": Object {
          "deletions": 2,
          "filesChanged": 1,
          "insertions": 0,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "3",
        "stats": Object {
          "deletions": 1,
          "filesChanged": 1,
          "insertions": 9,
        },
      },
    ],
    "stats": Object {
      "deletions": 4,
      "filesChanged": 3,
      "insertions": 9,
    },
  },
}
`;

exports[`commitsByDevelopers single git log result 1`] = `
Object {
  "kfiku.com@gmail.com": Object {
    "author": Object {
      "email": "kfiku.com@gmail.com",
      "name": "Grzegorz Klimek",
    },
    "commits": Array [
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "cleanup",
        "stats": Object {
          "deletions": 1,
          "filesChanged": 1,
          "insertions": 0,
        },
      },
    ],
    "stats": Object {
      "deletions": 1,
      "filesChanged": 1,
      "insertions": 0,
    },
  },
}
`;
